src/constraint/SequentialSampleable.cpp

```diff
@@ -0,0 +1,177 @@
+#include "aikido/constraint/SequentialSampleable.hpp"
+
+#include <dart/common/Console.hpp>
+#include <dart/common/Memory.hpp>
+
+namespace aikido {
+namespace constraint {
+
+//==============================================================================
+class SequentialSampleGenerator : public SampleGenerator
+{
+public:
+  /// Sample Generator for Sequential Sampleable
+  /// \param[in] stateSpace Statespace the associated constraint operates in
+  /// \param[in] generators Sequence of generators associated with corresponding
+  /// sequence of sampleables
+  SequentialSampleGenerator(
+      statespace::StateSpacePtr stateSpace,
+      std::vector<std::unique_ptr<SampleGenerator>> generators);
+
+  // Documentation inherited
+  statespace::StateSpacePtr getStateSpace() const override;
+
+  // Documentation inherited
+  bool sample(statespace::StateSpace::State* state) override;
+
+  // Documentation inherited
+  int getNumSamples() const override;
+
+  // Documentation inherited
+  bool canSample() const override;
+
+private:
+  /// StateSpace the associated sampleable operates in.
+  statespace::StateSpacePtr mStateSpace;
+
+  /// Sequence of generators associated with corresponding sequence of
+  /// sampleables.
+  std::vector<std::unique_ptr<SampleGenerator>> mGenerators;
+
+  /// Index of the active sampleable/generator.
+  std::size_t mIndex;
+};
+
+//==============================================================================
+SequentialSampleGenerator::SequentialSampleGenerator(
+    statespace::StateSpacePtr stateSpace,
+    std::vector<std::unique_ptr<SampleGenerator>> generators)
```

Comment:
Should this be passed by reference? (https://app.codacy.com/app/personalrobotics/aikido/pullRequest?prid=1615713)